sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4721,13 +4721,99 @@ def repeat(self, repeats):
       pd.core.strings.StringMethods, 'get_dummies',
       reason='non-deferred-columns')
 
-  split = frame_base.wont_implement_method(
-      pd.core.strings.StringMethods, 'split',
-      reason='non-deferred-columns')
+  def _split_helper(self, rsplit=False, **kwargs):
+    pat = kwargs.get('pat', None)
+    expand = kwargs.get('expand', False)
+    regex = kwargs.get('regex', None)
+
+    # regex introduced in pandas 1.4 but only for split, not
+    # rsplit, so removing it from kwargs
+    if PD_VERSION < (1, 4) and not rsplit:
+      kwargs.pop('regex', None)
+
+    if not expand:
+      # Not creating separate columns
+      proxy = self._expr.proxy()
+      if not rsplit:
+        func = lambda s: pd.concat([proxy, s.str.split(**kwargs)])
+      else:
+        func = lambda s: pd.concat([proxy, s.str.rsplit(**kwargs)])
+    else:
+      # Creating separate columns, so must be more strict on dtype
+      dtype = self._expr.proxy().dtype
+      if not isinstance(dtype, pd.CategoricalDtype):
+        method_name = 'rsplit' if rsplit else 'split'
+        raise frame_base.WontImplementError(
+            f"{method_name}() of non-categorical type is not supported because "
+            "the type of the output column depends on the data. Please use "
+            "pd.CategoricalDtype with explicit categories.",
+            reason="non-deferred-columns")
 
-  rsplit = frame_base.wont_implement_method(
-      pd.core.strings.StringMethods, 'rsplit',
-      reason='non-deferred-columns')
+      # Split (if applicable) the categories
+      if regex is False or (
+        regex is None and (
+          (not pat) or (isinstance(pat, str) and len(pat) == 1)
+        )
+      ):
+        # Treat pat as literal string
+        split_cats = [
+          cat.split(
+            sep=kwargs.get('pat'),
+            maxsplit=kwargs.get('n', -1)
+          ) for cat in dtype.categories
+        ]
+      else:
+        # Treat pat as regex
+        split_cats = [
+          re.split(
+            pattern=pat,
+            string=cat,
+            maxsplit=kwargs.get('n', 0)
+          ) for cat in dtype.categories
+        ]
```

Comment:
Actually, shouldn't we be able to get the categories just by calling pandas on the categories? Something like `dtype.categories.str.split(**kwargs)`

That would save us from needing to branch on the kwargs at all

Reply:
Silly me, I don't know why I overlooked that. Changed. Thanks!